pysd/py_backend/model.py

```diff
@@ -1690,33 +1710,329 @@ def _integrate(self, capture_elements):
         # need to add one more time step, because we run only the state
         # updates in the previous loop and thus may be one short.
         if self.time.in_return():
-            outputs.at[self.time.round()] = [getattr(self.components, key)()
-                                             for key in capture_elements]
+            out_obj.update(self)
 
         progressbar.finish()
 
-        # delete time column as it was created only for avoiding errors
-        # of appending data. See previous TODO.
-        del outputs["time"]
-        return outputs
 
-    def _add_run_elements(self, df, capture_elements):
+class OutputHandler():
+    """
+    Interface for the different output handlers.
+    """
+
+    def initialize(self, model, capture_elements):
+        pass
+
+    def update(self, model, capture_elements):
+        pass
+
+    def postprocess(self, **kwargs):
+        pass
+
+    def add_run_elements(self, model, capture_elements):
+        pass
```

Comment:
I think I would prefer having these methods with a "raise NotImplementedError"
I am thinking that probably it would be cleaner to move these Handlers to another file.

Reply:
Already done, but not pushed.